### PR TITLE
Introduce query for document ids with solver error

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## This introduces a breaking change

- [ ] Yes
- [x] No

## This should yield a new module release

- [x] Yes
- [ ] No

## This Pull Request implements

Introduce query to database for solver document IDs with errors

### CODE REVIEW

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -1094,6 +1094,63 @@ def get_error_solved_python_package_versions_all(
 
             return query.all()
 
+
+    def get_error_solved_document_id_all(
+        self,
+        package_name: str = None,
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:1100]:
```suggestion
        package_name: Optional[str] = None,
```

File: thoth/storages/graph/postgres.py [hunk 2]
```diff
@@ -1094,6 +1094,63 @@ def get_error_solved_python_package_versions_all(
...
+        package_version: str = None,
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:1101]:
```suggestion
        package_version: Optional[str] = None,
```

File: thoth/storages/graph/postgres.py [hunk 3]
```diff
@@ -1094,6 +1094,63 @@ def get_error_solved_python_package_versions_all(
...
+        index_url: str = None,
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:1102]:
```suggestion
        index_url: Optional[str] = None,
```

File: thoth/storages/graph/postgres.py [hunk 4]
```diff
@@ -1094,6 +1094,63 @@ def get_error_solved_python_package_versions_all(
...
+        *,
+        unsolvable: bool = False,
+        unparseable: bool = False,
+        start_offset: int = 0,
+        count: Optional[int] = DEFAULT_COUNT,
+        os_name: str = None,
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:1108]:
```suggestion
        os_name: Optional[str] = None,
```

File: thoth/storages/graph/postgres.py [hunk 5]
```diff
@@ -1094,6 +1094,63 @@ def get_error_solved_python_package_versions_all(
...
+        os_version: str = None,
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:1109]:
```suggestion
        os_version: Optional[str] = None,
```

File: thoth/storages/graph/postgres.py [hunk 6]
```diff
@@ -1094,6 +1094,63 @@ def get_error_solved_python_package_versions_all(
...
+        python_version: str = None,
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:1110]:
```suggestion
        python_version: Optional[str] = None,
```

File: thoth/storages/graph/postgres.py [hunk 7]
```diff
@@ -1094,6 +1094,63 @@ def get_error_solved_python_package_versions_all(
...
+        if unsolvable=True -> get_unsolvable_python_package_versions
+        if unparseable=True -> get_unparseable_python_package_versions
+
+        Examples:
+        >>> from thoth.storages import GraphDatabase
+        >>> graph = GraphDatabase()
+        >>> graph.get_error_solved_document_id_all()
+        ['solver-fedora-32-py37-324232']
+        """
+        os_version = OpenShift.normalize_os_version(os_name, os_version)
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:1125]: Normalization of `package_name` missing.

Comment (pacospace) [reply thoth/storages/graph/postgres.py:1125]: normalization is done inside `_construct_error_solved_python_package_versions_query`

Comment (fridex) [reply thoth/storages/graph/postgres.py:1125]: Ack :+1:

**Review by fridex — APPROVED** [review]
:+1:

**Comment by harshad16** [comment]
> /retest

the tests are failing due to package installation in pytest.
issue of boto3 installation.
```
Failed to install 'boto3', will try in next installation round...
Collecting boto3==1.14.53 (from -r /tmp/requirements.txt8xbhe5a5 (line 2))
  Downloading https://files.pythonhosted.org/packages/d5/5c/f891946d8e8adcccac4a4dacfab8976e5f1ec0ac5d1fb1f62343e6de34ac/boto3-1.14.53-py2.py3-none-any.whl (129kB)
THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    boto3==1.14.53 from https://files.pythonhosted.org/packages/d5/5c/f891946d8e8adcccac4a4dacfab8976e5f1ec0ac5d1fb1f62343e6de34ac/boto3-1.14.53-py2.py3-none-any.whl#sha256=defd1aa0bbc8eb9c3b6aefb5060972447f73a4f1c3ba995ba28d6f0b6b9059dd (from -r /tmp/requirements.txt8xbhe5a5 (line 2)):
        Expected sha256 b240ac281de363e25a8e1a4c862559d6a056d98dcb9f487fc94d73c6f6599dfc
             Got        defd1aa0bbc8eb9c3b6aefb5060972447f73a4f1c3ba995ba28d6f0b6b9059dd

Failed to install requirements, dependency 'boto3' could not be installed
```

**Comment by pacospace** [comment]
> > /retest
> 
> the tests are failing due to package installation in pytest.
> issue of boto3 installation.
> 
> ```
> Failed to install 'boto3', will try in next installation round...
> Collecting boto3==1.14.53 (from -r /tmp/requirements.txt8xbhe5a5 (line 2))
>   Downloading https://files.pythonhosted.org/packages/d5/5c/f891946d8e8adcccac4a4dacfab8976e5f1ec0ac5d1fb1f62343e6de34ac/boto3-1.14.53-py2.py3-none-any.whl (129kB)
> THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
>     boto3==1.14.53 from https://files.pythonhosted.org/packages/d5/5c/f891946d8e8adcccac4a4dacfab8976e5f1ec0ac5d1fb1f62343e6de34ac/boto3-1.14.53-py2.py3-none-any.whl#sha256=defd1aa0bbc8eb9c3b6aefb5060972447f73a4f1c3ba995ba28d6f0b6b9059dd (from -r /tmp/requirements.txt8xbhe5a5 (line 2)):
>         Expected sha256 b240ac281de363e25a8e1a4c862559d6a056d98dcb9f487fc94d73c6f6599dfc
>              Got        defd1aa0bbc8eb9c3b6aefb5060972447f73a4f1c3ba995ba28d6f0b6b9059dd
> 
> Failed to install requirements, dependency 'boto3' could not be installed
> ```

Should I update the Pipfile.lock?

**Comment by harshad16** [comment]
> Should I update the Pipfile.lock?

yes, that would be faster solution , or we can let kebechet fix it wit relock.

**Pull request merged**